src/Stream/MemoryStream.tpp

```diff
@@ -370,8 +329,103 @@ namespace Stream
       OutputMemoryStream(typename Allocator::size_type initial_size,
         const AllocatorInitializer& allocator_initializer)
       /*throw (eh::Exception)*/
-      : Holder(initial_size, allocator_initializer), Stream(this->buffer())
+      : Holder(initial_size, allocator_initializer)
+      , bad_(false)
+    {
+    }
+
+    template <typename Elem, typename Traits, typename Allocator,
+      typename AllocatorInitializer, const size_t SIZE>
+    void
+    OutputMemoryStream<Elem, Traits, Allocator, AllocatorInitializer, SIZE>::
+      append(Elem ch) /*throw (eh::Exception)*/
+    {
+      if (bad()) {
+        return;
+      }
+      auto* buffer = this->buffer();
+      if (buffer->ptr() == buffer->end() && !buffer->extend()) {
+        bad_ = true;
+        return;
+      }
+      *buffer->ptr() = ch;
+      buffer->incptr();
+    }
+
+    template <typename Elem, typename Traits, typename Allocator,
+      typename AllocatorInitializer, const size_t SIZE>
+    void
+    OutputMemoryStream<Elem, Traits, Allocator, AllocatorInitializer, SIZE>::
+      append(const Elem* str) /*throw (eh::Exception)*/
```

Comment:
Don't write throw specification if exception ins't possible here
Use noexcept if here no exceptions
or commented throw with list of possible exceptions

Reply:
use noexcept instead throw() - it is correct
throw() expressions present from time when we wrote code for old standard

Reply:
substituted throw() with noexcept in new code.